### PR TITLE
Prevent complex lambdas from being used in `.where(..., ...)`

### PR DESCRIPTION
**Pull request by dflemstr**
Builds on #21 due to the different way of extracting serialized lambdas.

The `serializeLambda` method was moved verbatim from a utility class to a private method since it only had one use site.

### CODE REVIEW

**Review by mattnworb — APPROVED** [review]
seems ok to me

**Comment by dflemstr** [comment]
Rebased

File: pojo/src/main/java/com/spotify/hamcrest/pojo/IsPojo.java [hunk 1]
```diff
@@ -138,4 +148,57 @@ private void indentDescription(Description description, Description innerDescrip
...
+        method.setAccessible(true);
+        return method;
+      });
+    } catch (PrivilegedActionException e) {
+      throw new IllegalStateException("Cannot serialize lambdas in unprivileged context", e);
+    }
+
+    try {
+      return (SerializedLambda) writeReplace.invoke(lambda);
+    } catch (IllegalAccessException | InvocationTargetException e) {
```
Comment (eshrubs) [inline pojo/src/main/java/com/spotify/hamcrest/pojo/IsPojo.java]: What about `ClassCastException`? Perhaps not needed, but I am certain that any `Serializable` will have `writeReplace` there. Maybe not, I am no java serialization expert

Comment (dflemstr) [reply pojo/src/main/java/com/spotify/hamcrest/pojo/IsPojo.java]: Good catch

**Pull request merged**